core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -836,64 +840,86 @@ private boolean containedInAdapter(NonBlockingStore nonBlockingStore, String ada
    }
 
    private boolean allowLoad(StoreStatus storeStatus, boolean localInvocation, boolean includeStores) {
-      return !storeStatus.characteristics.contains(Characteristic.WRITE_ONLY) && (localInvocation || !isLocalOnlyLoader(storeStatus.store)) &&
-             (includeStores || storeStatus.characteristics.contains(Characteristic.READ_ONLY) || storeStatus.config.ignoreModifications());
+      return !storeStatus.hasCharacteristic(Characteristic.WRITE_ONLY) &&
+             (localInvocation || !isLocalOnlyLoader(storeStatus.store)) &&
+             (includeStores || storeStatus.hasCharacteristic(Characteristic.READ_ONLY) ||
+              storeStatus.config.ignoreModifications());
    }
 
-   private boolean isLocalOnlyLoader(NonBlockingStore store) {
+   private boolean isLocalOnlyLoader(NonBlockingStore<?, ?> store) {
       if (store instanceof LocalOnlyCacheLoader) return true;
-      NonBlockingStore unwrappedStore;
+      NonBlockingStore<?, ?> unwrappedStore;
       if (store instanceof DelegatingNonBlockingStore) {
-         unwrappedStore = ((DelegatingNonBlockingStore) store).delegate();
+         unwrappedStore = ((DelegatingNonBlockingStore<?, ?>) store).delegate();
       } else {
          unwrappedStore = store;
       }
       if (unwrappedStore instanceof LocalOnlyCacheLoader) {
          return true;
       }
       if (unwrappedStore instanceof NonBlockingStoreAdapter) {
-         return ((NonBlockingStoreAdapter) unwrappedStore).getActualStore() instanceof LocalOnlyCacheLoader;
+         return ((NonBlockingStoreAdapter<?, ?>) unwrappedStore).getActualStore() instanceof LocalOnlyCacheLoader;
       }
       return false;
    }
 
    @Override
-   public CompletionStage<Long> size(Predicate<? super StoreConfiguration> predicate) {
+   public CompletionStage<Long> approximateSize(Predicate<? super StoreConfiguration> predicate, IntSet segments) {
       long stamp = acquireReadLock();
       try {
-         checkStoreAvailability();
-         if (log.isTraceEnabled()) {
-            log.tracef("Obtaining size from stores");
+         if (!isEnabled() || !isAvailable()) {
+            releaseReadLock(stamp);
+            return SIZE_UNAVAILABLE_FUTURE;
          }
-         NonBlockingStore<?, ?> nonBlockingStore = getStoreLocked(storeStatus -> storeStatus.characteristics.contains(
-               Characteristic.BULK_READ) && predicate.test(storeStatus.config));
-         if (nonBlockingStore == null) {
+
+         // Ignore stores without BULK_READ, they don't implement approximateSize()
+         StoreStatus firstStoreStatus = getStoreStatusLocked(storeStatus -> {
+            return storeStatus.hasCharacteristic(Characteristic.BULK_READ) &&
```

Comment:
Don't need brackets.

Reply:
Without brackets IntelliJ formats the code "the wrong way" (with my settings, at least, because I have `ij_java_align_multiline_parameters_in_calls = true` in my EditorConfig